local/local_test.go

```diff
@@ -927,131 +927,158 @@ func testLocalImage(t *testing.T, when spec.G, it spec.S) {
 				}
 			})
 
-			when("a single image name fails", func() {
-				it("returns results with errors for those that failed", func() {
-					failingName := newTestImageName() + ":🧨"
+			when("additional names are provided", func() {
+				var (
+					additionalRepoNames = []string{
+						repoName + ":" + h.RandString(5),
+						newTestImageName(),
+						newTestImageName(),
+					}
+					successfulRepoNames = append([]string{repoName}, additionalRepoNames...)
+				)
 
-					err := img.Save(append([]string{failingName}, additionalRepoNames...)...)
-					h.AssertError(t, err, fmt.Sprintf("failed to write image to the following tags: [%s:", failingName))
+				it.After(func() {
+					h.AssertNil(t, h.DockerRmi(dockerClient, additionalRepoNames...))
+				})
 
-					saveErr, ok := err.(imgutil.SaveError)
-					h.AssertEq(t, ok, true)
-					h.AssertEq(t, len(saveErr.Errors), 1)
-					h.AssertEq(t, saveErr.Errors[0].ImageName, failingName)
-					h.AssertError(t, saveErr.Errors[0].Cause, "invalid reference format")
+				it("saves to multiple names", func() {
+					h.AssertNil(t, img.Save(additionalRepoNames...))
 
 					for _, n := range successfulRepoNames {
-						_, _, err = dockerClient.ImageInspectWithRaw(context.TODO(), n)
+						_, _, err := dockerClient.ImageInspectWithRaw(context.TODO(), n)
 						h.AssertNil(t, err)
 					}
 				})
+
+				when("a single image name fails", func() {
+					it("returns results with errors for those that failed", func() {
+						failingName := newTestImageName() + ":🧨"
+
+						err := img.Save(append([]string{failingName}, additionalRepoNames...)...)
+						h.AssertError(t, err, fmt.Sprintf("failed to write image to the following tags: [%s:", failingName))
+
+						saveErr, ok := err.(imgutil.SaveError)
+						h.AssertEq(t, ok, true)
+						h.AssertEq(t, len(saveErr.Errors), 1)
+						h.AssertEq(t, saveErr.Errors[0].ImageName, failingName)
+						h.AssertError(t, saveErr.Errors[0].Cause, "invalid reference format")
+
+						for _, n := range successfulRepoNames {
+							_, _, err = dockerClient.ImageInspectWithRaw(context.TODO(), n)
+							h.AssertNil(t, err)
+						}
+					})
+				})
+			})
+		})
+
+		when("invalid image content for daemon", func() {
+			it("returns errors from daemon", func() {
+				var err error
+
+				repoName := newTestImageName()
+
+				invalidLayerTarFile, err := ioutil.TempFile("", "daemon-error-test")
+				h.AssertNil(t, err)
+				defer func() { invalidLayerTarFile.Close(); os.Remove(invalidLayerTarFile.Name()) }()
+
+				invalidLayerTarFile.Write([]byte("NOT A TAR"))
+				invalidLayerPath := invalidLayerTarFile.Name()
+
+				img, err := local.NewImage(repoName, dockerClient)
+				h.AssertNil(t, err)
+
+				err = img.AddLayer(invalidLayerPath)
+				h.AssertNil(t, err)
+
+				err = img.Save(repoName)
+				h.AssertError(t, err, fmt.Sprintf("failed to write image to the following tags: [%s:", repoName))
+				h.AssertError(t, err, "daemon response")
```

Comment:
Here's my attempt at some test coverage for this. I'm not sure how to repro the unclosed-response-body situation but this at least covers the sad-path of the daemon returning an error message but no golang `error`